Functionality for playing artist or album

andersthune opened the pull request:
As the dbus interface does not support playing a song in a context, the track search mode loses much of its functionality on linux. This PR adds functions for simply playing an artist or album from the track search mode. This is done either by calling one of the two functions `spotify-track-artist-select` or `spotify-track-album-select`, or by calling `spotify-track-select` (`M-RET`) while point is over a button representing an artist or album.

danielfm commented on spotify-track-search.el:
```diff
@@ -24,6 +24,20 @@
+	   (spotify-track-album-select))
+	  (t (spotify-track-select-default)))
+    )
+  )
```
I'd move these closing parenthesis to the end of the `(t (spotify-track-select-default)))`, as this is how the current code is styled.

This comment applies to other similar snippets of code.

andersthune replied:
Done.

andersthune commented:
Thanks for reviewing the request! I was still fairly new to elisp when I wrote the code, so I didn't know about the style conventions. I've adjusted the parentheses, and also rewritten `spotify-track-selected-button-type` in a more easy-to-read functional manner. I used some of the nice list functions from [dash](https://github.com/magnars/dash.el). If you'd prefer not having that as a dependency, I can rewrite it again using conventional `mapcar` etc, but I suspect the code would be less elegant.

andersthune commented:
I realized that I can just include a property with the buttons indicating whether the button is for an artist or album. This simplifies `spotify-track-selected-button-type` considerably, removing the need for an extra dependency.
